test/unit/domain/fiat/display-amounts-converter.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { DisplayAmountsConverter, DisplayCurrency } from "@domain/fiat"
+
+import { WalletCurrency, ZERO_CENTS, ZERO_SATS } from "@domain/shared"
+import { DisplayPriceRatio } from "@domain/payments"
+
+describe("DisplayAmountsConverter", () => {
+  const amounts: AmountsAndFees = {
+    btcPaymentAmount: { amount: 50_000n, currency: WalletCurrency.Btc },
+    btcProtocolAndBankFee: { amount: 1_000n, currency: WalletCurrency.Btc },
+    usdPaymentAmount: { amount: 1_250n, currency: WalletCurrency.Usd },
+    usdProtocolAndBankFee: { amount: 25n, currency: WalletCurrency.Usd },
+  }
+
+  const amountsAndZeroFees: AmountsAndFees = {
+    ...amounts,
+    btcProtocolAndBankFee: ZERO_SATS,
+    usdProtocolAndBankFee: ZERO_CENTS,
+  }
+
+  const btcQuoteAmount = {
+    amount: 5000n,
+    currency: WalletCurrency.Btc,
+  }
+
+  const displayQuoteAmount = {
+    amountInMinor: 100n,
+    displayInMajor: "1.00" as DisplayCurrencyMajorAmount,
+  }
+
+  const expectedResultForCurrency = (currency: DisplayCurrency) => {
+    // Based on ratio of `amounts` to `btcQuoteAmount`
+    const expectedResult = {
+      displayAmount: {
+        amountInMinor: 1000n,
+        currency: undefined,
+        displayInMajor: "10.00",
+      },
+      displayFee: { amountInMinor: 20n, currency: undefined, displayInMajor: "0.20" },
+      displayCurrency: undefined,
+      displayPrice: {
+        base: 20000000000n,
+        offset: 12n,
+        displayCurrency: undefined,
+        walletCurrency: "BTC",
+      },
+    }
+
+    return {
+      displayAmount: { ...expectedResult.displayAmount, currency },
+      displayFee: { ...expectedResult.displayFee, currency },
+      displayCurrency: currency,
+      displayPrice: { ...expectedResult.displayPrice, displayCurrency: currency },
+    }
+  }
+
+  const expectedResultForUsd = () => {
+    const expectedResult = expectedResultForCurrency(DisplayCurrency.Usd)
+    return {
+      ...expectedResult,
+      displayAmount: {
+        ...expectedResult.displayAmount,
+        amountInMinor: 1250n,
+        displayInMajor: "12.50",
+      },
+      displayFee: {
+        ...expectedResult.displayFee,
+        amountInMinor: 25n,
+        displayInMajor: "0.25",
+      },
+    }
+  }
+
+  describe("non-usd display currency", () => {
+    const currency = "EUR" as DisplayCurrency
```

Comment:
Usually ofr enums we use `DisplayCurrency.EUR` syntax.

Reply:
Related to https://github.com/GaloyMoney/galoy/pull/3135#discussion_r1320341780